playground/frontend/lib/components/split_view/split_view.dart

```diff
@@ -0,0 +1,175 @@
+
+  get _isVertical => widget.direction == SplitViewDirection.vertical;
+
+  @override
+  void initState() {
+    super.initState();
+    _ratio = widget.ratio;
+  }
+
+  @override
+  Widget build(BuildContext widgetContext) {
+    return LayoutBuilder(builder: (context, BoxConstraints constraints) {
+      _updateMaxSize(constraints);
+      return _isHorizontal
+          ? _buildHorizontalLayout(context, constraints)
+          : _buildVerticalLayout(context, constraints);
+    });
+  }
+
+  _buildHorizontalLayout(BuildContext context, BoxConstraints constraints) {
+    return SizedBox(
+      width: constraints.maxWidth,
+      child: Row(
+        children: <Widget>[
+          SizedBox(
+            width: _sizeFirst,
+            child: widget.first,
+          ),
+          _buildSeparator(context),
+          SizedBox(
+            width: _sizeSecond,
+            child: widget.second,
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildVerticalLayout(
+      BuildContext context, BoxConstraints constraints) {
+    return SizedBox(
+      height: constraints.maxHeight,
+      child: Column(
+        children: <Widget>[
+          SizedBox(
+            height: _sizeFirst,
+            child: widget.first,
+          ),
+          _buildSeparator(context),
+          SizedBox(
+            height: _sizeSecond,
+            child: widget.second,
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildSeparator(BuildContext context) {
+    return MouseRegion(
+      cursor: _isHorizontal
+          ? SystemMouseCursors.resizeLeftRight
+          : SystemMouseCursors.resizeUpDown,
+      child: GestureDetector(
+        behavior: HitTestBehavior.translucent,
+        child: Container(
+            width: _isHorizontal ? widget.dividerSize : double.infinity,
+            height: _isVertical ? widget.dividerSize : double.infinity,
+            color: ThemeColors.of(context).greyColor,
+            child: Center(
+              child: SvgPicture.asset(_isHorizontal
+                  ? kDragHorizontalIconAsset
+                  : kDragVerticalIconAsset),
+            )),
+        onPanUpdate: (DragUpdateDetails details) {
+          setState(() {
+            _updateRatio(details);
+          });
+        },
+      ),
+    );
+  }
+
+  void _updateRatio(DragUpdateDetails details) {
+    if (_isHorizontal) {
+      _ratio += details.delta.dx / _maxSize!;
+    } else {
+      _ratio += details.delta.dy / _maxSize!;
+    }
+    _checkRatioSafe();
+  }
+
+  void _checkRatioSafe() {
+    if (_ratio > maxRatio) {
+      _ratio = maxRatio;
+    }
+    if (_ratio < minRatio) {
+      _ratio = minRatio;
+    }
+  }
+
+  void _updateMaxSize(BoxConstraints constraints) {
+    _calculateMaxSize(
+        _isHorizontal ? constraints.maxWidth : constraints.maxHeight);
+  }
+
+  void _calculateMaxSize(double maxSize) {
+    _maxSize ??= maxSize - widget.dividerSize;
```

Comment:
This line is not needed, `maxSize` is a double so it cannot be null. The check underneath will always catch null values.